webhooky/tests/auth.rs

```diff
@@ -0,0 +1,741 @@
+    Ok(HttpResponseAccepted("ok".to_string()))
+}
+
+#[endpoint {
+    method = POST,
+    path = "/hmac/checkr/verify",
+}]
+async fn hmac_checkr_verification(
+    _rqctx: Arc<RequestContext<()>>,
+    body: HmacVerifiedBody<webhooky::handlers_checkr::CheckrWebhookVerification, CheckrWebhook>,
+) -> Result<HttpResponseAccepted<String>, HttpError> {
+    body.into_inner()?;
+    Ok(HttpResponseAccepted("ok".to_string()))
+}
+
+#[endpoint {
+    method = POST,
+    path = "/hmac/checkr/audit",
+}]
+async fn hmac_checkr_audit(
+    _rqctx: Arc<RequestContext<()>>,
+    body: HmacVerifiedBodyAudit<webhooky::handlers_checkr::CheckrWebhookVerification, CheckrWebhook>,
+) -> Result<HttpResponseAccepted<String>, HttpError> {
+    body.into_inner()?;
+    Ok(HttpResponseAccepted("ok".to_string()))
+}
+
+#[endpoint {
+    method = POST,
+    path = "/hmac/docusign/verify",
+}]
+async fn hmac_docusign_verification(
+    _rqctx: Arc<RequestContext<()>>,
+    body: HmacVerifiedBody<webhooky::handlers_docusign::DocusignWebhookVerification, Envelope>,
+) -> Result<HttpResponseAccepted<String>, HttpError> {
+    body.into_inner()?;
+    Ok(HttpResponseAccepted("ok".to_string()))
+}
+
+#[endpoint {
+    method = POST,
+    path = "/hmac/docusign/audit",
+}]
+async fn hmac_docusign_audit(
+    _rqctx: Arc<RequestContext<()>>,
+    body: HmacVerifiedBodyAudit<webhooky::handlers_docusign::DocusignWebhookVerification, Envelope>,
+) -> Result<HttpResponseAccepted<String>, HttpError> {
+    body.into_inner()?;
+    Ok(HttpResponseAccepted("ok".to_string()))
+}
+
+#[endpoint {
+    method = POST,
+    path = "/hmac/slack/verify",
+    content_type = "application/x-www-form-urlencoded"
+}]
+async fn hmac_slack_verification(
+    _rqctx: Arc<RequestContext<()>>,
+    body: HmacVerifiedBody<webhooky::handlers_slack::SlackWebhookVerification, BotCommand>,
+) -> Result<HttpResponseAccepted<BotCommand>, HttpError> {
+    Ok(HttpResponseAccepted(body.into_inner()?))
+}
+
+#[endpoint {
+    method = POST,
+    path = "/hmac/slack/audit",
+    content_type = "application/x-www-form-urlencoded"
+}]
+async fn hmac_slack_audit(
+    _rqctx: Arc<RequestContext<()>>,
+    body: HmacVerifiedBodyAudit<webhooky::handlers_slack::SlackWebhookVerification, BotCommand>,
+) -> Result<HttpResponseAccepted<BotCommand>, HttpError> {
+    Ok(HttpResponseAccepted(body.into_inner()?))
+}
+
+#[endpoint {
+    method = POST,
+    path = "/hmac/slack/interactive/verify",
+    content_type = "application/x-www-form-urlencoded"
+}]
+async fn hmac_slack_interactive_verification(
+    _rqctx: Arc<RequestContext<()>>,
+    body: HmacVerifiedBody<webhooky::handlers_slack::SlackWebhookVerification, InteractiveEvent>,
+) -> Result<HttpResponseAccepted<slack_chat_api::InteractivePayload>, HttpError> {
+    let event = body.into_inner()?;
+    let ser_payload = urlencoding::decode(&event.payload).unwrap();
+    let payload: slack_chat_api::InteractivePayload = serde_json::from_str(&ser_payload).unwrap();
```

Comment:
I guess this is just the format they send, but doubly-URL-encoded JSON? Really? Gah.